systems/subscriber/subscriber-registry/pkg/server/subscriber.go

```
@@ -0,0 +1,288 @@
+
+	return resp, nil
+
+}
+
+func (s *SubcriberServer) ListSubscribers(ctx context.Context, req *pb.ListSubscribersRequest) (*pb.ListSubscribersResponse, error) {
+
+	if err := ctx.Err(); err != nil {
+		return nil, err
+	}
+
+	logrus.Infof("List all subscribers")
+
+	subscribers, err := s.subscriberRepo.ListSubscribers()
+	if err != nil {
+		logrus.WithError(err).Error("error while getting all subscribers")
+		return nil, grpc.SqlErrorToGrpc(err, "subscribers")
+	}
+
+	subscriberList := &pb.ListSubscribersResponse{
+		Subscribers: dbsubscriberToPbSubscribers(subscribers),
+	}
+
+	return subscriberList, nil
+}
+func (s *SubcriberServer) GetByNetwork(ctx context.Context, req *pb.GetByNetworkRequest) (*pb.GetByNetworkResponse, error) {
+	networkID := req.GetNetworkID()
+	if networkID == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "networkID must not be empty")
+	}
+
+	if err := ctx.Err(); err != nil {
+		return nil, err
+	}
+
+	logrus.Infof("Get subscribers by network: %v ", networkID)
+	networkIDUUID, err := uuid.FromString(networkID)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid networkID: %v", err)
+	}
+
+	subscribers, err := s.subscriberRepo.GetByNetwork(networkIDUUID)
+	if err != nil {
+		logrus.WithError(err).Error("error while getting subscribers by network")
+		return nil, grpc.SqlErrorToGrpc(err, "subscribers")
+	}
+
+	subscriberList := &pb.GetByNetworkResponse{
+		Subscribers: dbsubscriberToPbSubscribers(subscribers),
+	}
+
+	return subscriberList, nil
+}
+func (s *SubcriberServer) Update(ctx context.Context, req *pb.UpdateSubscriberRequest) (*pb.UpdateSubscriberResponse, error) {
+	subscriberIdReq := req.GetSubscriberID()
+	if subscriberIdReq == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "subscriberID must not be empty")
+	}
+
+	if err := ctx.Err(); err != nil {
+		return nil, err
+	}
+	subscribeIDUUID, err := uuid.FromString(subscriberIdReq)
```

1) subscriberIDUUID does not make sense.
2) err is being swallowed.